code/modules/cm_marines/equipment/sentries.dm

```diff
@@ -931,16 +1016,45 @@
 		var/mob/living/carbon/human/H = M
 		if(istype(H) && H.get_target_lock(iff_signal)) continue
 
+
+
+		if(alerts_on)
+			if(world.time > (last_alert + SENTRY_ALERT_DELAY) || !(M in alert_list)) //if we're not on cooldown or the target isn't in the list, sound the alarm
+				playsound(loc, 'sound/machines/warning-buzzer.ogg', 50, FALSE)
+				sentry_alert(SENTRY_ALERT_HOSTILE, M)
+				alert_list.Add(M)
+				last_alert = world.time
+
 		var/angle = get_dir(src, M)
-		if(angle & dir)
-			path = getline2(src, M)
+		if(angle & dir || radial_mode)
+			path = getline2(src, M, TRUE)
 		else
 			continue
 
 		if(path.len)
+			var/blocked = FALSE //reset blocked
 			for(T in path)
-				if(T.opacity) continue
-			targets += M
+				if(T.opacity || T.density)
```

Comment:
Won't this prevent firing over barricades?

Reply:
Fixed